homeassistant/helpers/condition.py

```diff
@@ -820,9 +820,15 @@ def time(
                 after_entity.attributes.get("minute", 59),
                 after_entity.attributes.get("second", 59),
             )
-        elif after_entity.attributes.get(
-            ATTR_DEVICE_CLASS
-        ) == SensorDeviceClass.TIMESTAMP and after_entity.state not in (
+        elif after_entity.domain == "time":
+            try:
+                after = datetime.strptime(after_entity.state, "%H:%M:%S").time()
+            except ValueError:
+                return False
```

Comment:
Why do we return `False` here? When does a time entity have a state which can't be parsed?
Should we instead check the state is not in (unavailable, unknown) as for timestamp sensors; if it is return `False`, and if the state can't be parsed raise `ConditionError`?

Reply:
It's to let the unit test pass: 
`hass.states.async_set("time.invalid_timestamp", "This is not a timestamp")`

If the state becomes invalid, I don't think we should raise?

Reply:
Right, but calling `hass.states.async_set` with a value the time entity can't have in a test is not a good test IMO.

Based in the time entity's implementation, it's state can be:
- `unavailable` if the time entity's available property is `False`
- `unknown` if the the time entity's `native_value` is `None`
- a valid time string

Reply:
I've removed the invalid test case, and added a test case for unavailable and unknown specifically.